code/src/components/Loading.js

```diff
@@ -0,0 +1,46 @@
+/* eslint-disable react/jsx-no-useless-fragment */
+import React, { useState, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import { Player } from '@lottiefiles/react-lottie-player';
+import styled from 'styled-components/macro';
+
+export const Loading = () => {
+  const isLoading = useSelector((state) => state.labyrinth.isLoading);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    setLoading(isLoading);
+  }, [isLoading]);
+  return (
```

Comment:
To control the duration of the animation, we added this code after the last dispatch in our thunk(s): 
setTimeout(() => { dispatch(loading.actions.setLoading(false)) }, 1500). You can also put the speed at 1.0 to slow your animation down a bit more.